polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -279,6 +281,21 @@ impl Queue {
 
 		let job = queue.remove(job_index).expect("Job is just checked to be in queue; qed");
 
+		if let Some(deadline) = job.exec_deadline {
+			let now = Instant::now();
+			gum::debug!(target: LOG_TARGET, ?priority, ?deadline, ?now, "Job has a deadline");
+			if now > deadline {
+				let _ = job.result_tx.send(Err(ValidationError::ExecutionDeadline));
+				gum::warn!(
+					target: LOG_TARGET,
+					?priority,
+					?deadline,
+					"Job exceeded its deadline and was dropped without execution",
+				);
+				return
```

Comment:
try_assign_next gets called when a job finishes or when something else gets enqueued in the work queue, returning here, it means we will miss a scheduling opportunity, shouldn't we actually try to assign the next job ?